routes/tasks.js

```diff
@@ -216,6 +187,47 @@ router.put('/:task_id', (req, res, next) => {
   });
 });
 
+router.put('/:task_id/complete', (req, res, next) => {
+  const task_id = ObjectID(req.params.task_id);
+  console.log(`task id: ${task_id}`);
+  req.app.locals.tasks.findOneAndUpdate(
+    { _id: task_id },
+    {
+      $set: {
+        isDone: true
+      }
+    },
+    {
+      projection: { "mangoTransactions": 1, "user_id": 1, "isDone": 1 },
+    }
+  ).then((updatedTask) => {
+    //TODO: Consider ways to ensure atomicity of these chained operations
+    console.log("findone and update document");
+    console.dir(updatedTask);
+    const { mangoTransactions, user_id, isDone } = updatedTask.value;
+    if (isDone) {
+      res.status(503).end();
+      return;
+    }
+    const mangosEarned = sumMangos(mangoTransactions);
+    res.status(200).send({mangosEarned});
+    // console.log(`sum mangos: ${mangosEarned}`);
```

Comment:
Are you keeping this commented code for future?

Reply:
nope, i'll clean that up; forgot to delete it